guides/workflow/testing.md

```diff
@@ -0,0 +1,185 @@
+
+These testing frameworks let you define expected behaviors for your code in different scenarios and then compare them with the actual outcomes.
+
+In the sections below, we'll show you how to set up specific testing frameworks with Wix CLI.
+
+## Vitest
+
+Vitest is a blazing fast unit test framework powered by Vite ([Learn more about Vitest](https://vitest.dev/guide/why.html)).
+
+Following Vitest's [getting started guide](https://vitest.dev/guide/), you can start by installing Vitest:
+
+```
+yarn install -D vitest
+```
+
+Also, install `jsdom` which we'll need for simulating a browser-like environment:
+
+```
+yarn install -D jsdom
+```
+
+Then, create the following `vitest.config.mjs` file at the root of your application:
+
+```javascript
+import { defineConfig } from 'vitest/config'
+
+export default defineConfig({
+  test: {
+    environment: 'jsdom',
+  },
+});
+```
+
+Next, add the following script to your `package.json`:
+
+```diff
+{
+    "scripts": {
++     "test": "vitest"
+    }
+}
+```
+
+We recommend you use [Testing Library](https://testing-library.com/) for testing your UI components. To set it up, start by installing it:
+
+```
+yarn install -D @testing-library/jest-dom @testing-library/react
+```
+
+Then, add the following to your `vitest.config.mjs`:
+
+```diff
+import { defineConfig } from 'vitest/config'
+
+export default defineConfig({
+  test: {
+    environment: 'jsdom',
++   setupFiles: ['./src/test-setup.ts'],
+  },
+});
+```
+
+Finally, create `src/test-setup.ts` which adds [Testing Library](https://testing-library.com/)'s matchers:
+
+```typescript
+import '@testing-library/jest-dom/vitest';
+```
+
+Once all this has been set up, here's a example test for a dashboard page:
+
+```tsx
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import { expect, it } from 'vitest';
+import Page from './page';
+
+it('renders', () => {
+  render(<Page />, { legacyRoot: true });
+  expect(screen.getByText('Page Title')).toBeInTheDocument();
+});
+```
+
+## Jest
+
+[Jest](https://jestjs.io/) is a delightful JavaScript Testing Framework with a focus on simplicity.
+
+To set it up, start by installing its relevant dependencies:
+
+```
+yarn install -D jest jest-environment-jsdom jsdom identity-obj-proxy @swc/core @swc/jest
+```
+
+Then, create the following `jest.config.js` file at the root of your application:
+
+```javascript
+/** @type {import('jest').Config} */
+module.exports = {
+  testEnvironment: "jsdom",
+  moduleNameMapper: {
+    '^(?!.*(\\.st\\.css))(\\.{1,2}/.*)\\.js$': '$2',
+    '\\.(css|scss)$': 'identity-obj-proxy',
+  },
+  extensionsToTreatAsEsm: ['.ts', '.tsx'],
+  transform: {
+    '^.+\\.(mt|t|cj|j)s?x$': ['@swc/jest'],
+    '\\.(jpg|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$':
+      '<rootDir>/__mocks__/fileMock.js',
```

Comment:
Notice that I added this (as well as the file it references to below) to handle images and static files. I couldn't find a proper package that does this (like `identity-obj-proxy`) other than https://github.com/dferber90/jest-transform-file.